### PR TITLE
wannier90: init at 3.1.0

### PR DESCRIPTION
**Pull request by mathunje**

### CODE REVIEW

File: pkgs/apps/wannier90/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  ];
+
+  src = fetchFromGitHub {
+    owner = "wannier-developers";
+    repo = "wannier90";
+    rev = "v${version}";
+    hash = "sha256-+Mq7lM6WuwAnK/2FlDz9gNRIg2sRazQRezb3BfD0veY=";
+  };
+
+  postPatch = ''
```
Comment (sheepforce) [inline pkgs/apps/wannier90/default.nix:20]: Maybe you could put a comment here why those tests were removed.

File: pkgs/apps/wannier90/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  doCheck = true;
+  checkInputs = [ python2 ];
+  checkTarget = [ "test-serial" ];
+
+
+  hardeningDisable = [ "format" ];
+
+  meta = with lib; {
+    description = "Wannierization program, currently provided only serial (as parallel tests fail)";
```
Comment (sheepforce) [inline pkgs/apps/wannier90/default.nix:42]:
```suggestion
    Calculation of maximally localised Wannier functions
```

File: pkgs/apps/wannier90/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    homepage = "http://www.wannier.org";
+    license = licenses.gpl2;
```
Comment (sheepforce) [inline pkgs/apps/wannier90/default.nix:44]:
```suggestion
    license = licenses.gpl2Plus;
```

**Review by sheepforce — CHANGES_REQUESTED** [review]

File: pkgs/apps/wannier90/default.nix (hunk 2 above)
Comment (sheepforce) [reply pkgs/apps/wannier90/default.nix]: Ah sorry, of course `description = ` and the quotations marks have to be kept. :)

File: pkgs/apps/wannier90/default.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  doCheck = true;
+  checkInputs = [ python2 ];
+  checkTarget = [ "test-serial" ];
+
+
+  hardeningDisable = [ "format" ];
+
+  meta = with lib; {
+    Calculation of maximally localised Wannier functions
```
Comment (markuskowa) [inline pkgs/apps/wannier90/default.nix:43]:
```suggestion
    description = "Calculation of maximally localised Wannier functions";
```

File: result [hunk 5]
```diff
@@ -0,0 +1 @@
+/nix/store/x6dq4ay8cmfg21bcqkbnsd0ckagm5vwb-wannier90-3.1.0
```
Comment (markuskowa) [inline result:1]: Please remove the result link from the git tracking.

File: pkgs/apps/wannier90/default.nix [hunk 6]
```diff
@@ -0,0 +1,49 @@
+{ stdenv, lib, gfortran, blas, lapack, python2, fetchFromGitHub }:
+
```
Comment (markuskowa) [inline pkgs/apps/wannier90/default.nix:2]: I assume that the package can only handle 4 byte integers. Let's ensure we build it with the correct API.
```suggestion
assert (!blas.isILP64);
assert blas.isILP64 == lapack.isILP64;

```

**Review by markuskowa — CHANGES_REQUESTED** [review]

**Comment by markuskowa** [comment]
Please also squash your commits, once you are finished.

File: pkgs/apps/wannier90/default.nix [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    cp config/make.inc.gfort make.inc
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp -p wannier90.x postw90.x $out/bin
+  '';
+
+  doCheck = true;
+  checkInputs = [ python2 ];
```
Comment (markuskowa) [inline pkgs/apps/wannier90/default.nix:38]: One last thing. Just tests it: it seems that pyton3 just works fine here:
```suggestion
  checkInputs = [ python3 ];
```

**Review by markuskowa — APPROVED** [review]
Thanks!

**Pull request merged**